### PR TITLE
Implement and test a CUDA version of the module map

### PR DESCRIPTION
**Pull request by stephenswat**
The module map was originally designed to be GPU friendly, and we now reap the fruits of that decision. This implementation of a CUDA version of that code is quite short and simple to understand. This pull request adds an owning CUDA version of the module map, as well as a light-weight non-owning view object which is designed to be passed to kernels. It also tests the class.

This pull request depends on #153.

### CODE REVIEW

**Comment by stephenswat** [comment]
This PR also depends on https://github.com/acts-project/vecmem/pull/172.

**Comment by stephenswat** [comment]
This is now ready to be merged.

File: core/include/traccc/geometry/module_map.hpp [hunk 1]
```diff
@@ -15,6 +15,11 @@
 #include <iostream>
```
Comment (beomki-yeo) [inline core/include/traccc/geometry/module_map.hpp:15]: I think you can remove it as well

Comment (beomki-yeo) [inline core/include/traccc/geometry/module_map.hpp:15]: I think you can remove it as well

**Review by beomki-yeo — COMMENTED** [review]
Just very trivial comments...

**Comment by stephenswat** [comment]
Thanks! You have a keen eye for this! :smile:

**Comment by stephenswat** [comment]
Closing this as there seems to be little interest in it.

**Comment by beomki-yeo** [comment]
Actually I need it for GPU. 
I was not event aware of that this PR has existed so I made a (very simple) module map for CKF study recently.
So I guess it is worth keeping this alive

**Comment by stephenswat** [comment]
Mind you this is a map from module identifiers to transformation matrices, although we can easily repurpose this for spacepoint identifier to module identifier mappings, which I guess is what you need for the CKF.

**Comment by beomki-yeo** [comment]
> although we can easily repurpose this for spacepoint identifier to module identifier mappings,

Exactly. I hope we can generalize it for multiple purposes.

**Comment by krasznaa** [comment]
Right now you could still argue that we don't have the correct thing in place yet. But as I wrote in https://github.com/acts-project/traccc/pull/592#discussion_r1618846697, I'll have some EDM updates in the foreseeable future, which will absolutely replace this PR.

So yeah, I agree with closing it. Even if I was only going to suggest it a little later, once I pushed my EDM changes through. :wink:

**Pull request closed without merging**